src/main/java/com/sun/service/HistoryService.java

```diff
@@ -1,22 +1,84 @@
 package com.sun.service;
 
+import com.sun.dto.HistoryDto;
+import com.sun.dto.HistoryRequestDto;
+import com.sun.entity.Book;
 import com.sun.entity.History;
+import com.sun.entity.HistoryDetail;
+import com.sun.entity.User;
+import com.sun.mapper.HistoryMapper;
+import com.sun.repository.HistoryDetailRepository;
 import com.sun.repository.HistoryRepository;
+import com.sun.repository.UserRepository;
 import org.springframework.stereotype.Service;
 import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.ArrayList;
 import java.util.List;
+import java.util.logging.Logger;
 import javax.transaction.Transactional;
 
 @Service
 @Transactional
 public class HistoryService {
+    Logger logger = Logger.getLogger(HistoryService.class.getName());
     @Autowired
     private HistoryRepository historyRepository;
-    public List<History> getAll(){
-        return  historyRepository.findAll();
+    @Autowired
+    private HistoryDetailRepository historyDetailRepository;
+    @Autowired
+    private UserRepository userRepository;
+
+    public List<History> getAll() {
+        return historyRepository.findAll();
     }
 
-    public History findById(Integer id){
+    public History findById(Integer id) {
         return historyRepository.findById(id).orElse(null);
     }
+
+    public List<HistoryDto> getHistory(int userId) {
+        List<HistoryDto> historyDtos = new ArrayList<>();
+        try {
+            List<History> histories = historyRepository.findByUserId(userId);
+            if (!histories.isEmpty()) {
```

Comment:
chị thấy đâu cần check if này em nhỉ, vì em dùng dạng foreach, nếu `histories` mà empty thì ko bao giờ nó chạy vào block trong for đâu nè